internal/clients/http/azure/create_vm.go

```diff
@@ -0,0 +1,398 @@
+							ID: subnet.ID,
+						},
+						PublicIPAddress: &armnetwork.PublicIPAddress{
+							ID: publicIP.ID,
+						},
+					},
+				},
+			},
+			NetworkSecurityGroup: &armnetwork.SecurityGroup{
+				ID: nsg.ID,
+			},
+		},
+	}
+
+	pollerResponse, err := nicClient.BeginCreateOrUpdate(ctx, resourceGroupName, name, parameters, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := pollerResponse.PollUntilDone(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return &resp.Interface, err
+}
+
+func (c *client) createVirtualMachine(ctx context.Context, location string, resourceGroupName string, instanceType armcompute.VirtualMachineSizeTypes, networkInterface *armnetwork.Interface, imageID string, sshKeyBody string, vmName string, diskName string) (*armcompute.VirtualMachine, error) {
+	ctx, span := otel.Tracer(TraceName).Start(ctx, "createVirtualMachine")
+	defer span.End()
+
+	vmClient, err := c.newVirtualMachinesClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	parameters := armcompute.VirtualMachine{
+		Location: to.Ptr(location),
+		Identity: &armcompute.VirtualMachineIdentity{
+			Type: to.Ptr(armcompute.ResourceIdentityTypeNone),
+		},
+		Properties: &armcompute.VirtualMachineProperties{
+			StorageProfile: &armcompute.StorageProfile{
+				ImageReference: &armcompute.ImageReference{
+					ID: ptr.To(imageID),
+				},
+				OSDisk: &armcompute.OSDisk{
+					Name:         to.Ptr(diskName),
+					CreateOption: to.Ptr(armcompute.DiskCreateOptionTypesFromImage),
+					Caching:      to.Ptr(armcompute.CachingTypesReadWrite),
+					ManagedDisk: &armcompute.ManagedDiskParameters{
+						StorageAccountType: to.Ptr(armcompute.StorageAccountTypesStandardLRS), // OSDisk type Standard/Premium HDD/SSD
+					},
+					//DiskSizeGB: to.Ptr[int32](100), // default 127G
+				},
+			},
+			HardwareProfile: &armcompute.HardwareProfile{
+				VMSize: to.Ptr(instanceType), // VM size include vCPUs,RAM,Data Disks,Temp storage.
+			},
+			OSProfile: &armcompute.OSProfile{ //
+				ComputerName:  to.Ptr(vmName),
+				AdminUsername: to.Ptr(adminUsername),
+				//AdminPassword: to.Ptr("Password01!@#"),
+				//require ssh key for authentication on linux
+				LinuxConfiguration: &armcompute.LinuxConfiguration{
+					DisablePasswordAuthentication: to.Ptr(true),
+					SSH: &armcompute.SSHConfiguration{
+						PublicKeys: []*armcompute.SSHPublicKey{
+							{
+								Path:    to.Ptr(fmt.Sprintf("/home/%s/.ssh/authorized_keys", adminUsername)),
+								KeyData: to.Ptr(sshKeyBody),
+							},
+						},
+					},
+				},
+			},
+			NetworkProfile: &armcompute.NetworkProfile{
+				NetworkInterfaces: []*armcompute.NetworkInterfaceReference{
+					{
+						ID: networkInterface.ID,
+					},
+				},
+			},
+		},
+	}
+
+	pollerResponse, err := vmClient.BeginCreateOrUpdate(ctx, resourceGroupName, vmName, parameters, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := pollerResponse.PollUntilDone(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I think this is reasonable for the first version. More info: https://learn.microsoft.com/en-us/azure/developer/go/azure-sdk-core-concepts